src/parser.rs

```diff
@@ -59,106 +104,528 @@ impl Header {
     }
 }
 
-/// Parses the different kinds of SomeIP messages (at the moment only basic type) as Result.
-#[deprecated(note = "use Someip::from_slice instead")]
-pub fn someip_message(input: &[u8]) -> Result<Message, Error> {
-    Message::from_slice(input)
-}
+impl SdPayload {
+    pub fn from_slice(s: &[u8]) -> Result<SdPayload, Error> {
+        if s.len() < SD_HEADER_LENGTH {
+            return Err(Error::NotEnoughData {
+                min: SD_HEADER_LENGTH,
+                actual: s.len(),
+            });
+        }
```

Comment:
Is this a optimisation? Think that every SD message has payload. Is that correct?

Reply:
correct!